### PR TITLE
Add Pgp signing template

### PR DESCRIPTION
**Pull request by TravisEz13**
Add Pgp signing template
* make it clear which json template ran
* log the generated json

### CODE REVIEW

File: template-compliance/authenticode-sign.yml [hunk 1]
```diff
@@ -56,5 +56,5 @@ steps:
     $vstsCommandString = "vso[task.setvariable variable=GDN_CODESIGN_TARGETDIRECTORY]${{ parameters.signOutputPath }}"
     Write-Verbose -Message ("sending " + $vstsCommandString) -Verbose
     Write-Host "##$vstsCommandString"
-  displayName: Generate signing JSON
+  displayName: Generate Authenticode signing JSON
   condition: and(and(and(succeeded(), eq(variables['Build.Reason'], 'Manual')), ne(variables['SkipSigning'], 'True')), ne(variables['SigningServer'], ''))
```
Comment (JamesWTruher) [inline template-compliance/authenticode-sign.yml:60]: this seems to be the same calculation as in EsrpSign. Is there someway to have only a single instance of this calculation be shared between files?

File: EsrpSign.yml [hunk 2]
```diff
@@ -73,5 +87,5 @@ steps:
     Write-Host "##vso[artifact.upload containerfolder=signingReport;artifactname=signingReport]$(Agent.TempDirectory)\$fileName"
     Copy-Item -Path ${{ parameters.buildOutputPath }}\* -Dest ${{ parameters.signOutputPath }}\ -Recurse -Force -Verbose
```
Comment (adityapatwardhan) [inline EsrpSign.yml:88]: Should the copy happen before the ESRP task so we can have unsigned files still in buildOutputPath?

File: template-compliance/pgp-sign.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+parameters:
+  - name: "buildOutputPath"
```
Comment (adityapatwardhan) [inline template-compliance/pgp-sign.yml:2]: This parameter is not used in this file.

File: template-compliance/pgp-sign.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    $vstsCommandString = "vso[task.setvariable variable=$VariableName][$($esrp | ConvertTo-Json -Compress)]"
+    Write-Verbose -Message ("sending " + $vstsCommandString) -Verbose
+    Write-Host "##$vstsCommandString"
+
+    $vstsCommandString = "vso[task.setvariable variable=GDN_CODESIGN_TARGETDIRECTORY]${{ parameters.signOutputPath }}"
+    Write-Verbose -Message ("sending " + $vstsCommandString) -Verbose
+    Write-Host "##$vstsCommandString"
+  displayName: Generate PGP signing JSON
+  condition: and(and(and(succeeded(), eq(variables['Build.Reason'], 'Manual')), ne(variables['SkipSigning'], 'True')), ne(variables['SigningServer'], ''))
```
Comment (JamesWTruher) [inline template-compliance/pgp-sign.yml:37]: yep - and here is that calculation again

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by JamesWTruher — APPROVED** [review]
i think this looks fine

File: template-compliance/pgp-sign.yml (hunk 4 above)
Comment (TravisEz13) [reply template-compliance/pgp-sign.yml:37]: unfortunately, we tried doing the calculation one time and it causes issues

File: template-compliance/pgp-sign.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
+parameters:
+  - name: "buildOutputPath"
+    default: "$(Build.ArtifactStagingDirectory)\\build"
```
Comment (TravisEz13) [inline template-compliance/pgp-sign.yml:3]:
```suggestion
```

File: template-compliance/authenticode-sign.yml (hunk 1 above)
Comment (TravisEz13) [reply template-compliance/authenticode-sign.yml:60]: The best I can think of is to pass it in as parameter and it causes issues.  I'll file a feature request for step templates to be conditional.

https://developercommunity.visualstudio.com/idea/1262508/allow-specifying-a-condition-for-a-set-of-template.html

File: EsrpSign.yml (hunk 2 above)
Comment (TravisEz13) [reply EsrpSign.yml:88]: This is a behavior change.  I can work on this after I get this work done.

I think this makes sense

Comment (TravisEz13) [reply EsrpSign.yml:88]: One thing, is we might end up with unsigned files in the signed folder.

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**